pages/src/rtl-switcher/rtl-switcher.ts

```diff
@@ -0,0 +1,79 @@
+import {ready} from '../../../src/modules/esl-utils/decorators/ready';
+import {attr} from '../../../src/modules/esl-base-element/decorators/attr';
+import {bind} from '../../../src/modules/esl-utils/decorators/bind';
+import {boolAttr, ESLBaseElement} from '../../../src/modules/esl-base-element/core';
+
+import {TraversingQuery} from '../../../src/modules/all';
+
+export class ESLDemoSwitcher extends ESLBaseElement {
+  static is = 'esl-d-rtl-switcher';
+
+  protected _$target: any;
+
+  @boolAttr({readonly: true}) public active: boolean;
+
+  @attr({defaultValue: '::next'}) public target: string;
+
+  @ready
+  protected connectedCallback() {
+    super.connectedCallback();
+    this.updateTargetFromSelector();
+    this.setInitialDirection();
+  }
+
+  @ready
+  protected disconnectedCallback() {
+    super.disconnectedCallback();
+  }
+
+  public get $target(): any {
+    return this._$target;
+  }
+
+  public set $target(newPopupInstance) {
+    this._$target = newPopupInstance;
+    this.bindEvents();
+  }
+
+  protected updateTargetFromSelector() {
+    if (!this.target) return;
+    this.$target = TraversingQuery.first(this.target, this);
+  }
+
+  protected bindEvents() {
+    this.addEventListener('click', this._onClick);
+  }
+
+  protected unbindEvents() {
+    this.removeEventListener('click', this._onClick);
+  }
+
+  protected setInitialDirection() {
+    this.ltrDirection();
+  }
+
+  public toggleDirection(state: boolean = !this.active) {
+    state ? this.rtlDirection() : this.ltrDirection();
+  }
+
+  public ltrDirection() {
+    this.$target.setAttribute('dir', 'ltr');
+  }
+
+  public rtlDirection() {
+    this.$target.setAttribute('dir', 'rtl');
+  }
+
+  @bind
+  protected _onClick(event: MouseEvent) {
+    event.preventDefault();
+    this.toggleDirection();
+    this.updateState();
+  }
+
+  public updateState() {
+    const isActive = this.$target?.dir === 'rtl' ? true : false;
```

Comment:
Can be simplified

```suggestion
    const isActive = this.$target?.dir === 'rtl';
```

And also it can be moved to separated getter.